### ISSUE TITLE
Adding Enclosure Switch Open/Closed

### ISSUE THREAD

**Issue by Lasermax3000** [makerbase-mks/MKS-DLC32#134]
Hello MKS-Team,

this shall not be a new Issue, even more a enhancement in Safety. 
I would like to know if there is any free IO-Port (may be SDA PIN) on the DLC32 to connect a switch for detecting if a Laser or CNC-Machine Enclosure is open or closed. So that you can push the Start Engraving Button and the Engraver waits until the Enclosure is closed before starting the engraving process. And Additionally when you open the enclosure it interrupts the e.g. Engraving till the Enclosure gets closed, as a Safety Feature.
Please let me know if you got any Ideas, how the realise this function.

**Comment by abinoer** [comment]
All IO ports are already in use, there is no extra IO to implement this

**Comment by Lasermax3000** [comment]
Sad to hear, but why can´t the SDA Pin be used? I think it actually is only used to get in Flashing Mode at Startup. 
Have you another idea how to realise this function? Maybe with the Reset Button or Probe Switch?
Thank you for your support.

**Comment by abinoer** [comment]
No, we can't. It'll interfere with other functions. I'm sorry
